src/gax-internal/src/grpc.rs

```diff
@@ -182,7 +182,7 @@ impl Client {
     }
 
     async fn make_inner(endpoint: Option<String>, default_endpoint: &str) -> Result<InnerClient> {
-        let endpoint = tonic::transport::Endpoint::from_shared(
+        let endpoint = tonic::transport::Endpoint::new(
```

Comment:
`::new()` is an undocumented feature:

https://github.com/hyperium/tonic/blob/50253f1cb236feca3061488d542e37cf60ba4f65/tonic/src/transport/channel/endpoint.rs#L54-L57


We should not be using it. That is why I changed the code to use `from_shared()`, but maybe we need to do something more difficult?

Reply:
> `::new()` is an undocumented feature:

Ah, I missed this. Thanks.

> something more difficult?

Fixed by copying the implementation of `new()`.

(I am not going to update the instances of `Endpoint::new()` in generated code in this PR. Eventually the generated code will use the `grpc::Client` instead.)